### PR TITLE
Docs: Add a router or multiplexer

### LINKED ISSUE #66401: Epic: Improve backend plugins documentation
The documentation for [backend plugins](https://grafana.com/docs/grafana/latest/developers/plugins/backend/)  is quite sparse. It does however give good high level information, but doesn't answer how it works on a more technical level. Would be nice with some additional technical details, like architectural figures and so on. 

In addition, there are some backend plugin documentation mixed in with the [Add authentication for data source plugins](https://grafana.com/docs/grafana/latest/developers/plugins/add-authentication-for-data-source-plugins/#authenticate-using-a-backend-plugin). Might be an opportunity to divide the [Build a plugin](https://grafana.com/docs/grafana/latest/developers/plugins/) to make it more clear what's related to plugins without a backend and the other way around.

```[tasklist]
### Tasks
- [x] https://github.com/grafana/grafana/issues/75706 https://github.com/grafana/plugin-tools/pull/448
- [ ] https://github.com/grafana/grafana/issues/79862
- [ ] https://github.com/grafana/plugin-tools/issues/480
- [ ] New page: Instance management
- [ ] New page: Error handling (related https://github.com/grafana/grafana/issues/75761)
- [ ] New page: Add a resource handler (https://community.grafana.com/t/how-to-add-a-resource-handler-for-your-data-source/58924)
- [ ] New page: Add a query multiplexer/router (https://community.grafana.com/t/how-to-add-a-query-multiplexer-router-for-your-datasource/59406)
- [ ] New page: HTTP client
```

### PR DESCRIPTION
**Pull request by josmperez**
How to add a router or multiplexer to your plugin. Draft topic based on a [community forum post](https://community.grafana.com/t/how-to-add-a-query-multiplexer-router-for-your-datasource/59406).

Fixes [#66401 subtask](https://github.com/grafana/grafana/issues/66401)

### CODE REVIEW

**Review by sunker — APPROVED** [review]
Looks good to me! I'm not sure we need to refer to the Azure Monitor examples. The BE for this DS is a little messy and also it's moving target so might become out of date. The inline snippets are descriptive enough IMO.

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Normally you implement the `QueryData` method in your backend plugin for data queries. But what if you need to support different kinds of queries: metrics, logs, and traces, for instance? That’s where the usage of a query _router_ (also known as a _multiplexer_) comes handy.
+
+The plugin development requirement is that you need to populate the `queryType` property of your query model client-side, see the [`DataQuery`](https://github.com/grafana/grafana/blob/a728e9b4ddb6532b9fa2f916df106e792229e3e0/packages/grafana-data/src/types/query.ts#L47) interface.
+
+With `queryType` populated in queries and sent to your backend plugin below is an example of how you would use the [`datasource.QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) to multiplex or route different query types to separate query handlers.
+
+Implemented in this way, each query handler can then `json.Unmarshal` each query JSON field in [`DataQuery`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend#DataQuery) to a certain Go struct as shown in this example:
+
+```
```
Comment (marefr) [inline docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md:23]:
```suggestion
```go
```

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+description: Add a router or multiplexer to your plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - router
+  - multiplexer
+---
+
+# Add a router or multiplexer to your plugin
```
Comment (marefr) [inline docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md:13]: This is specific for data source with backend. Becomes pretty long though

```suggestion
# Add a query router or multiplexer to your data source backend
```

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+// handleQueryFallback handle queries without a matching query type handler registered.
+func (d *MyDatasource) handleQueryFallback(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+```
+
+## Advanced usage
+
+An example of using [`QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) can be found for Grafana's built-in Grafana Azure Monitor data source. Refer to this code for examples of implementation:
```
Comment (marefr) [inline docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md:81]: We don't cover this in https://github.com/grafana/grafana-plugin-examples yet, but now we have the testdata datasource using it so linking to that.

```suggestion
An example of using [`QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) can be found for Grafana's built-in TestData data source. Refer to this code for examples of implementation:
```

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+- [grafana/azuremonitor.go](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L49) at 164ce63e280565cc960fb0e0d9f074ec003d5a2a - See the code at the [grafana/grafana](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L49) repo.
+- [grafana/azuremonitor.go](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L182-L201) at 164ce63e280565cc960fb0e0d9f074ec003d5a2a - See the code at the [grafana/grafana](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L182-L201) repo.
+- [grafana/azuremonitor.go](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L55-L57) at 164ce63e280565cc960fb0e0d9f074ec003d5a2a - See the code at the [grafana/grafana](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L55-L57) repo.
```
Comment (marefr) [inline docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md:85]:
```suggestion
- [create query type multiplexer](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/testdata.go#L22) and [calls registerScenarios](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/testdata.go#L44)
- [registerScenarios method](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/scenarios.go#L33) uses a [helper method](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/scenarios.go#L204-L207) to register each query type handler. The latter also shows how you can wrap the actual handler in another handler to apply common functionality/middleware to all handlers, e.g. logging and traces. 
```

**Review by marefr — APPROVED** [review]
LGTM great. Some  minor comments/suggestions

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md (hunk 4 above)
Comment (marefr) [reply docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md:85]: @josmperez seems like you missed to incorporate this suggestion, but added my earlier suggestion?

**Pull request merged**